tests/utils/LHNTestUtils.tsx

```diff
@@ -281,18 +282,29 @@ function getFakeAdvancedReportAction(actionName: ActionName = 'IOU', actor = 'em
 function MockedSidebarLinks({currentReportID = ''}: MockedSidebarLinksProps) {
     return (
         <ComposeProviders components={[OnyxProvider, LocaleContextProvider, EnvironmentProvider, CurrentReportIDContextProvider]}>
-            <SidebarLinksData
-                // @ts-expect-error TODO: Remove this once SidebarLinksData (https://github.com/Expensify/App/issues/25220) is migrated to TypeScript.
-                onLinkClick={() => {}}
-                insets={{
-                    top: 0,
-                    left: 0,
-                    right: 0,
-                    bottom: 0,
-                }}
-                isSmallScreenWidth={false}
-                currentReportID={currentReportID}
-            />
+            {/*
+             * Only required to make unit tests work, since we
+             * explicitly pass the currentReportID in LHNTestUtils
+             * to SidebarLinksData, so this context doesn't have an
+             * access to currentReportID in that case.
+             *
+             * So this is a work around to have currentReportID available
+             * only in testing environment.
+             *  */}
+            <ReportIDsContextProvider currentReportIDForTests={currentReportID}>
```

Comment:
can you create a context here instead for the currentReportID?

Reply:
No we can't. I tried doing that but since both `currentReportID` and `reportIDs` are contexts, we don't have the updated value between the contexts when initial value is being passed as props.

Reply:
does `SidebarLinksData` take `currentReportID` or does it only take the prop when being tested?

Reply:
`SidebarlinksData` takes `currentReportID` from the props in both cases. In testing, we are explicitly sending the `currentReportID` and in real scenarios, `SidebarLinksData` is wrapped with `withCurrentReportID`, which feeds the prop. 

Your question might be if we already have `withCurrentReportiD` in `SidebarLinksData` then why we are explicitly setting `currentReportID` in testing?

To answer it, the Reassure tests are different then an e2e tests. In Reassure, we are testing the single component in isolation, which means there's no navigation. And since there's no navigation, we don't have any `currentReportiD` updated and that's why we have `currentReportID` manually set.

Reply:
okay I think I now kind of understand this. thank you for the explanation really 🙇